packages/core-data/src/entities.js

```diff
@@ -351,6 +359,7 @@ async function loadPostTypeEntities() {
 				}/${ parentId }/revisions${
 					revisionId ? '/' + revisionId : ''
 				}`,
+			revisionKey: isTemplate ? 'wp_id' : DEFAULT_ENTITY_KEY,
```

Comment:
I kind of expected this logic to be within `getRevisionsUrl` instead of actually changing the "key" used in state...

Reply:
Is there another way to grab the unique revision id for templates?

`getRevisionsUrl` builds the revisions URL, with the `/<parent>/revisions/<id>` format.

`revisionKey` was supposed to centralize the logic to grab the revision entity key.

Reply:
No there's no other way, but I guess in my mind, the use of revisionKey should be within the function rather than passing the correct id as argument. 

I guess in mind, the "store"/"reducer"... should only use the regular "id" and only when calling the API we should make use of "wp_id". Does that make sense?

Reply:
I'm not sure I get it, but it is late for me 😆 It would be nicer to call the API using the _real_ WP id, e.g., 

`getRevisions( 'postType', 'wp_template', 22 )`

instead of

`getRevisions( 'postType', 'wp_template', 'twentytwentyfour//home' )`

That would mean changing the controller in Core I suppose.

I tried to do something similar to the `entityConfig.key` pattern, and also to not touch the reducer at all.  

It's unfortunate gymnastics just to appease the way templates rest controllers work, but I wasn't sure how else to approach it. 🤷🏻

Reply:
I see... so we'd fetch the parent record via `select` in the resolver to get its `id`, or `wp_id` in the case of templates and template parts.

Pity the template revisions controller doesn't accept ids as well. Maybe, to be complete and easier to work with (no regex magic) it should?

The corollary issue is that we still need to store revision records by a unique id, so somehow the reducer needs to know what record key to work with to achieve the following state:

```js
[
1: {
    wp_id: 1, id: 'theme//template_name_y', ...revisionData, 
},
2: {
    wp_id: 2, id: 'theme//template_name_y', ...revisionData, 
},
...
]
```

Reply:
> The corollary issue is that we still need to store revision records by a unique id, so somehow the reducer needs to know what record key to work with to achieve the following state:

The id is unique, there's no need to store wp_ids there.

Reply:
> The id is unique, there's no need to store wp_ids there.

I think we might be talking about different things. 

I'm talking exclusively about template revision items from `WP_REST_Template_Revisions_Controller`, whose `id` keys are not unique in relation to other revisions of the same parent:

<img width="1171" alt="Screenshot 2023-11-29 at 7 22 03 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/83d0a157-354a-4757-a4a3-358bf7419e39">

Reply:
Oh yeah, there's something weird in this response for me. This call is to retrieve revisions. So I would expect the "id" to correspond to the revision "id" not the "template" id. I expect parent to actually contain that highlighted value and "id" to contain "wp_id" (if that's the revision id). In which case I think we should update the endpoint.

Reply:
For the post revisions endpoint it's like that https://developer.wordpress.org/rest-api/reference/post-revisions/

Reply:
> I expect parent to actually contain that highlighted value and "id" to contain "wp_id" (if that's the revision id). In which case I think we should update the endpoint

For sure. I can only guess revisions are that way to make reinstating them easier 🤷🏻